packages/clay-alert/src/ClayAlert.tsx

```diff
@@ -0,0 +1,177 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import * as React from 'react';
+import classNames from 'classnames';
+import Icon from '@clayui/icon';
+
+const useAutoClose = (autoClose?: boolean | number, onClose = () => {}) => {
+	const timer = React.useRef<number | undefined>(undefined);
+
+	let cancelTimer = () => {};
+	let startTimer = () => {};
+
+	if (autoClose) {
+		cancelTimer = () => {
+			if (timer.current) {
+				clearTimeout(timer.current);
+
+				timer.current = undefined;
+			}
+		};
+
+		startTimer = () => {
+			timer.current = setTimeout(
+				onClose,
+				autoClose === true ? 8000 : autoClose
```

Comment:
By default we should leave the timeout in 5s and we should freeze the timer when you have a `hover` on alert.

Reply:
I had gotten the default value of 8 seconds from [here](https://github.com/liferay/clay/blob/2.x-develop/packages/clay-alert/src/ClayToast.soy#L16).

And I went ahead with canceling and restarting the timer on hover since it seemed weird that if a user hovers with only 200ms remaining, it would close super quick. I thought it would be a little more user friendly to clear and restart on hover. What do you think?

Reply:
No problems over the 8 seconds, let's keep do not remember why we changed to that value.

The bad thing is that if you have 200ms missing and it restarts it will stay for a lot longer, I do not know which would be the best case for that, we have pros and cons in both use cases.

@victorvalle can you take a look at this?

Reply:
We choose 8s as a good value reading other guides.
I don't understand why it restarts sorry. When you are hovering it the timer should freeze.

Reply:
I was going to say the "freeze" thing too. As a user, I expect the thing to stick around indefinitely if I am hovering over it, but I *don't* expect the timer to reset. ie. if I start hovering with 2 seconds left on the timer, and I move my mouse away 30 seconds later, I expect the toast to disappear 2 seconds after my mouse leaves.

Reply:
👍 ill go ahead and make it a pause rather than reset.